spec/src/consensus.rs

```diff
@@ -44,6 +45,10 @@ impl ProposalWindow {
     pub fn start(&self) -> BlockNumber {
```

Comment:
From proposing block's perspective, `block_number + self.0` is the `start` that its proposals become effective, and `block_number + self.1` is the `end` after which its proposals become out dated.

But from the current block's perspective, `block_number - self.1` is the `start` which proposals can commit, and `block_number - self.1` is the `end`.

The function name `start`, `end` and the response type `BlockNumber` also make me think they will return the block number of the start and end block, but indeed they return the offsets.

I'll start a new thread for this, lets just ignore it in the PR since it is not introduced here.